## Final Project Submission

Please fill out:
* Student name: Eugene Asengi Ukiru
* Student pace: Full time 
* Instructor name: Maryann Mwikali


In [622]:
# Your code here - remember to use markdown cells for comments as well!

IMPORTING RELEVANT LIBRARIES

In [623]:
# Run ipython magic command to display matplotlib graphs
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [624]:
# Loading data of Aviation accidents
Aviation_data = pd.read_csv('data/Aviation_Data.csv', low_memory=False)
Aviation_data

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude   Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2        Saltville, VA  United States  36.922223  -81.878056          NaN   
3           EUREKA, CA  United States        NaN         NaN          NaN   
4           Canton, OH  United States        NaN         NaN          NaN   
...                ...            ...        ...         ...          ...   
90343    Annapolis, MD  United States        NaN         NaN          NaN   
90344      Hampton, NH  United States        NaN         NaN          NaN   
90345       Payson, AZ  United States    341525N    1112021W          PAN   
90346       Morgan, UT  United States        NaN         NaN          NaN   
90347       Athens, GA  United States        NaN         NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0    

In [625]:
Aviation_data.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

Data Understanding  

The DataFrame provides me with a preview of how the data seems to be. I can see what to expect when I begin to dig deeper. Inorder to get full knowledge and insight on what the information contains, I will use the pandas function **.info()** and **.shape**. It is a crucial step for it fabricates  a base for the next step which is data cleaning.

In [626]:
Aviation_data.shape

(90348, 31)

In [627]:
Aviation_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

There are 90347 rows and 31 columns in the dataframe.

In [628]:
# Checking the list of columns.
Aviation_data.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

The function .columns provides names of all column names in the dataframe.

Statistical Summary.

In [629]:
# The summary statistics of numeric columns
Aviation_data.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

**Data Preparation.**  
In this section, we will prepare the data for the analysis through cleaning and organizing, handling missing values, filtering irrelevant columns hence making the data accurate.

In [630]:
# Checking boolean values of missing values present in each column
Aviation_data.isna().any()

Event.Id                   True
Investigation.Type        False
Accident.Number            True
Event.Date                 True
Location                   True
Country                    True
Latitude                   True
Longitude                  True
Airport.Code               True
Airport.Name               True
Injury.Severity            True
Aircraft.damage            True
Aircraft.Category          True
Registration.Number        True
Make                       True
Model                      True
Amateur.Built              True
Number.of.Engines          True
Engine.Type                True
FAR.Description            True
Schedule                   True
Purpose.of.flight          True
Air.carrier                True
Total.Fatal.Injuries       True
Total.Serious.Injuries     True
Total.Minor.Injuries       True
Total.Uninjured            True
Weather.Condition          True
Broad.phase.of.flight      True
Report.Status              True
Publication.Date           True
dtype: b

Use of the function .isna().sum() to confirm the total number of NaN values in the entire DataFrame.

In [631]:
# Checking the number of missing values in each column
Aviation_data.isna().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40099
Airport.Name              37558
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2776
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8536
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7840
Publication.Date          16689
dtype: i

Finding overall % of missing values in columns.  
This assists in establishing which columns have the highest number of missing values and the impact they have in my analysis.

In [632]:
# Finding missing in column(%)
Aviation_data.isna().mean()*100

Event.Id                   1.614867
Investigation.Type         0.000000
Accident.Number            1.614867
Event.Date                 1.614867
Location                   1.672422
Country                    1.865011
Latitude                  61.944924
Longitude                 61.954886
Airport.Code              44.382831
Airport.Name              41.570372
Injury.Severity            2.721698
Aircraft.damage            5.150086
Aircraft.Category         64.263736
Registration.Number        3.072564
Make                       1.684597
Model                      1.716695
Amateur.Built              1.727764
Number.of.Engines          8.348829
Engine.Type                9.447913
FAR.Description           64.555939
Schedule                  86.073848
Purpose.of.flight          8.468367
Air.carrier               81.573471
Total.Fatal.Injuries      14.233851
Total.Serious.Injuries    15.461327
Total.Minor.Injuries      14.822686
Total.Uninjured            8.158454
Weather.Condition          6

In [633]:
Aviation_data_copy = Aviation_data.copy()

In [634]:
Aviation_data = Aviation_data_copy.copy()

In [635]:
Aviation_data.drop(columns=['Schedule', 'FAR.Description', 'Latitude', 'Longitude', 'Location', 'Air.carrier', 'Airport.Name'], inplace = True)

In [636]:
print(Aviation_data.isna().mean()*100)
print(Aviation_data.shape)

Event.Id                   1.614867
Investigation.Type         0.000000
Accident.Number            1.614867
Event.Date                 1.614867
Country                    1.865011
Airport.Code              44.382831
Injury.Severity            2.721698
Aircraft.damage            5.150086
Aircraft.Category         64.263736
Registration.Number        3.072564
Make                       1.684597
Model                      1.716695
Amateur.Built              1.727764
Number.of.Engines          8.348829
Engine.Type                9.447913
Purpose.of.flight          8.468367
Total.Fatal.Injuries      14.233851
Total.Serious.Injuries    15.461327
Total.Minor.Injuries      14.822686
Total.Uninjured            8.158454
Weather.Condition          6.586753
Broad.phase.of.flight     31.681941
Report.Status              8.677558
Publication.Date          18.471909
dtype: float64
(90348, 24)


In [637]:
Aviation_data.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [638]:
# Replacing missing values with the median
Aviation_data_columns = ['Number.of.Engines', 'Total.Fatal.Injuries', 'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured']
Aviation_data[Aviation_data_columns] = Aviation_data[Aviation_data_columns].fillna(Aviation_data[Aviation_data_columns].median())

In [639]:
Aviation_data.isna().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Country                    1685
Airport.Code              40099
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2776
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines             0
Engine.Type                8536
Purpose.of.flight          7651
Total.Fatal.Injuries          0
Total.Serious.Injuries        0
Total.Minor.Injuries          0
Total.Uninjured               0
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7840
Publication.Date          16689
dtype: int64

In [640]:
# Replacement of all missing categorical values
Aviation_data.fillna('Missing', inplace= True)


In [641]:
Aviation_data.isna().sum()

Event.Id                  0
Investigation.Type        0
Accident.Number           0
Event.Date                0
Country                   0
Airport.Code              0
Injury.Severity           0
Aircraft.damage           0
Aircraft.Category         0
Registration.Number       0
Make                      0
Model                     0
Amateur.Built             0
Number.of.Engines         0
Engine.Type               0
Purpose.of.flight         0
Total.Fatal.Injuries      0
Total.Serious.Injuries    0
Total.Minor.Injuries      0
Total.Uninjured           0
Weather.Condition         0
Broad.phase.of.flight     0
Report.Status             0
Publication.Date          0
dtype: int64

In [642]:
# Dropping duplicates
Aviation_data.drop_duplicates()
Aviation_data

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

             Country Airport.Code Injury.Severity Aircraft.damage  \
0      United States      Missing        Fatal(2)       Destroyed   
1      United States      Missing        Fatal(4)       Destroyed   
2      United States      Missing        Fatal(3)       Destroyed   
3      United States      Missing        Fatal(2)       Destroyed   
4      United States      Missing        Fatal(1)       Destroyed   
...              ...          ...             ...             ...   
90343  United States      Missing           Minor         Missing   
90344  United States      Missing         Missing         Missing   
90345  United States          PAN       Non-Fatal     Substantial   
90346  United States      Missing         Missing         Missing   
90347  United States      Missing           Minor         Missing   

      Aircraft.Category Registration.Number  ...    Engine.Type  \
0               Missing              NC6404  ...  Reciprocating   
1               Missing              N5069P  ...  Reciprocating   
2               Missing              N5142R  ...  Reciprocating   
3               Missing              N1168J  ...  Reciprocating   
4               Missing               N15NY  ...        Missing   
...                 ...                 ...  ...            ...   
90343           Missing              N1867H  ...        Missing   
90344           Missing              N2895Z  ...        Missing   
90345          Airplane              N749PJ  ...        Missing   
90346           Missing              N210CU  ...        Missing   
90347           Missing              N9026P  ...        Missing   

      Purpose.of.flight Total.Fatal.Injuries  Total.Serious.Injuries  \
0              Personal                  2.0                     0.0   
1              Personal                  4.0                     0.0   
2              Personal                  3.0                     0.0   
3              Personal                  2.0                     0.0   
4              Personal                  1.0                     2.0   
...                 ...                  ...                     ...   
90343          Personal                  0.0                     1.0   
90344           Missing                  0.0                     0.0   
90345          Personal                  0.0                     0.0   
90346          Personal                  0.0                     0.0   
90347          Personal                  0.0                     1.0   

      Total.Minor.Injuries Total.Uninjured  Weather.Condition  \
0                      0.0             0.0                UNK   
1                      0.0             0.0                UNK   
2                      0.0             1.0                IMC   
3                      0.0             0.0                IMC   
4                      0.0             0.0                VMC   
...                    ...             ...                ...   
90343                  0.0             0.0            Missing   
90344                  0.0             0.0            Missing   
90345                  0.0             1.0                VMC   

**Aggregation methods and .groupby()**  
In this section, we will explore the different aggregation methods and how to use the .groupby()
This helps in understanding different relationships between what causes accidents and what is the most aircraft prone to failure. Grouping the data using the groupby() method is highly beneficial especially for categorical variables that may provide insight into how different factors influence injuries, fatalities and other metrics.

In [643]:
Aviation_data.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Country', 'Airport.Code', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type',
       'Purpose.of.flight', 'Total.Fatal.Injuries', 'Total.Serious.Injuries',
       'Total.Minor.Injuries', 'Total.Uninjured', 'Weather.Condition',
       'Broad.phase.of.flight', 'Report.Status', 'Publication.Date'],
      dtype='object')

In [644]:
Aviation_data

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

             Country Airport.Code Injury.Severity Aircraft.damage  \
0      United States      Missing        Fatal(2)       Destroyed   
1      United States      Missing        Fatal(4)       Destroyed   
2      United States      Missing        Fatal(3)       Destroyed   
3      United States      Missing        Fatal(2)       Destroyed   
4      United States      Missing        Fatal(1)       Destroyed   
...              ...          ...             ...             ...   
90343  United States      Missing           Minor         Missing   
90344  United States      Missing         Missing         Missing   
90345  United States          PAN       Non-Fatal     Substantial   
90346  United States      Missing         Missing         Missing   
90347  United States      Missing           Minor         Missing   

      Aircraft.Category Registration.Number  ...    Engine.Type  \
0               Missing              NC6404  ...  Reciprocating   
1               Missing              N5069P  ...  Reciprocating   
2               Missing              N5142R  ...  Reciprocating   
3               Missing              N1168J  ...  Reciprocating   
4               Missing               N15NY  ...        Missing   
...                 ...                 ...  ...            ...   
90343           Missing              N1867H  ...        Missing   
90344           Missing              N2895Z  ...        Missing   
90345          Airplane              N749PJ  ...        Missing   
90346           Missing              N210CU  ...        Missing   
90347           Missing              N9026P  ...        Missing   

      Purpose.of.flight Total.Fatal.Injuries  Total.Serious.Injuries  \
0              Personal                  2.0                     0.0   
1              Personal                  4.0                     0.0   
2              Personal                  3.0                     0.0   
3              Personal                  2.0                     0.0   
4              Personal                  1.0                     2.0   
...                 ...                  ...                     ...   
90343          Personal                  0.0                     1.0   
90344           Missing                  0.0                     0.0   
90345          Personal                  0.0                     0.0   
90346          Personal                  0.0                     0.0   
90347          Personal                  0.0                     1.0   

      Total.Minor.Injuries Total.Uninjured  Weather.Condition  \
0                      0.0             0.0                UNK   
1                      0.0             0.0                UNK   
2                      0.0             1.0                IMC   
3                      0.0             0.0                IMC   
4                      0.0             0.0                VMC   
...                    ...             ...                ...   
90343                  0.0             0.0            Missing   
90344                  0.0             0.0            Missing   
90345                  0.0             1.0                VMC   

In [645]:
# Grouping by Aircraft category and calculate mean number of accidents
df = Aviation_data.groupby('Aircraft.Category').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df

Aircraft.Category  Total.Fatal.Injuries  Total.Serious.Injuries  \
0            Airplane              0.580403                0.262954   
1             Balloon              0.186147                0.805195   
2               Blimp              0.000000                0.000000   
3              Glider              0.194882                0.222441   
4           Gyrocraft              0.254335                0.317919   
5          Helicopter              0.516860                0.302326   
6             Missing              0.552677                0.217065   
7   Powered Parachute              0.164835                0.439560   
8        Powered-Lift              0.000000                0.200000   
9              Rocket              1.000000                0.000000   
10               ULTR              0.000000                0.000000   
11                UNK              0.000000                0.000000   
12         Ultralight              0.333333                0.366667   
13            Unknown              1.142857                0.500000   
14               WSFT              1.111111                0.111111   
15       Weight-Shift              0.416149                0.360248   

    Total.Minor.Injuries  Total.Uninjured  
0               0.227613         6.192237  
1               0.800866         3.653680  
2               0.750000         1.250000  
3               0.220472         0.722441  
4               0.173410         0.520231  
5               0.259302         1.165116  
6               0.341606         4.698989  
7               0.802198         0.252747  
8               0.000000         0.800000  
9               1.000000         0.000000  
10              1.000000         0.000000  
11              0.000000         0.000000  
12              0.266667         0.466667  
13              0.071429         1.285714  
14              0.222222         0.111111  
15              0.310559         0.291925

In [647]:
df = Aviation_data[Aviation_data['Aircraft.Category'] == 'Airplane'].groupby('Make').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df


Make  Total.Fatal.Injuries  Total.Serious.Injuries  \
0               177MF LLC              0.000000                2.000000   
1     2007 Savage Air LLC              0.000000                0.000000   
2             2021FX3 LLC              0.000000                0.000000   
3                  3XTRIM              1.000000                0.000000   
4            5 RIVERS LLC              0.000000                0.000000   
...                   ...                   ...                     ...   
3870                 Zlin              0.000000                0.000000   
3871        Zlin Aviation              0.000000                0.000000   
3872               Zodiac              1.000000                0.000000   
3873         de Havilland              0.222222                0.444444   
3874              unknown              0.000000                0.000000   

      Total.Minor.Injuries  Total.Uninjured  
0                 0.000000         0.000000  
1                 0.000000         4.000000  
2                 0.000000         2.000000  
3                 1.000000         0.000000  
4                 1.000000         1.000000  
...                    ...              ...  
3870              0.000000         1.000000  
3871              0.000000         1.000000  
3872              1.000000         0.000000  
3873              0.111111         5.777778  
3874              1.000000         0.000000  

[3875 rows x 5 columns]

In [648]:
df = Aviation_data.groupby('Country').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df

Country  Total.Fatal.Injuries  Total.Serious.Injuries  \
0    ATLANTIC OCEAN              3.296296                0.370370   
1                AY              0.000000                0.000000   
2       Afghanistan              9.142857                0.785714   
3           Albania              0.000000                0.000000   
4           Algeria             53.500000                0.500000   
..              ...                   ...                     ...   
215     West Indies              0.363636                0.090909   
216        Wolseley              1.000000                0.000000   
217           Yemen              0.000000                0.000000   
218          Zambia              5.500000                0.000000   
219        Zimbabwe              1.000000                0.333333   

     Total.Minor.Injuries  Total.Uninjured  
0                3.209877        57.839506  
1                0.000000         0.000000  
2                0.214286        15.214286  
3                0.000000         0.000000  
4                0.000000         0.500000  
..                    ...              ...  
215              1.545455         1.818182  
216              0.000000         0.000000  
217              0.000000         0.000000  
218              0.000000         0.500000  
219              0.333333         7.666667  

[220 rows x 5 columns]

In [649]:
df = Aviation_data.groupby('Model').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df

Model  Total.Fatal.Injuries  Total.Serious.Injuries  \
0                   &GCBC                   0.0                     0.0   
1               (EX) RV-6                   2.0                     0.0   
2              (MODIFIED)                   1.0                     0.0   
3      (SOLOY CONVERSION)                   0.0                     0.0   
4                   -269C                   0.0                     0.0   
...                   ...                   ...                     ...   
12314               lebed                   2.0                     0.0   
12315                none                   1.0                     1.0   
12316           sportstar                   0.0                     0.0   
12317                 unk                   0.0                     1.0   
12318             unknown                   1.0                     0.0   

       Total.Minor.Injuries  Total.Uninjured  
0                       0.0              2.0  
1                       0.0              0.0  
2                       0.0              0.0  
3                       0.0              1.0  
4                       0.0              1.0  
...                     ...              ...  
12314                   0.0              0.0  
12315                   0.0              1.0  
12316                   0.0              2.0  
12317                   1.0              0.0  
12318                   0.0              1.0  

[12319 rows x 5 columns]

In [653]:
df=Aviation_data.groupby('Engine.Type').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df

Engine.Type  Total.Fatal.Injuries  Total.Serious.Injuries  \
0          Electric              0.200000                0.100000   
1   Geared Turbofan              0.000000                0.000000   
2     Hybrid Rocket              1.000000                0.000000   
3                LR              0.000000               10.500000   
4           Missing              1.534677                0.325211   
5              NONE              0.000000                0.000000   
6              None              0.052632                0.631579   
7     Reciprocating              0.340026                0.193528   
8         Turbo Fan              1.837969                0.744861   
9         Turbo Jet              1.226174                0.335704   
10       Turbo Prop              0.757299                0.199646   
11      Turbo Shaft              0.469659                0.337213   
12              UNK              0.000000                0.000000   
13          Unknown              1.838128                0.552413   

    Total.Minor.Injuries  Total.Uninjured  
0               0.000000         0.700000  
1               0.000000        10.083333  
2               1.000000         0.000000  
3               1.500000         5.500000  
4               0.254100        10.439784  
5               0.000000         1.000000  
6               0.421053         4.736842  
7               0.249259         1.143794  
8               1.364369        85.091495  
9               0.519203        48.516358  
10              0.265998         5.044235  
11              0.433084         1.425326  
12              2.000000         0.000000  
13              0.852755         6.298879

In [654]:
df = Aviation_data.groupby('Make').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df

Make  Total.Fatal.Injuries  Total.Serious.Injuries  \
0     107.5 Flying Corporation              1.000000                0.000000   
1                         1200              0.000000                1.000000   
2                    177MF LLC              0.000000                2.000000   
3             1977 Colfer-chan              0.000000                0.000000   
4                   1st Ftr Gp              1.000000                0.000000   
...                        ...                   ...                     ...   
8233                  Zukowski              0.000000                0.000000   
8234                     Zwart              0.000000                0.000000   
8235              de Havilland              0.222222                0.444444   
8236                     drone              0.000000                0.000000   
8237                   unknown              0.000000                0.000000   

      Total.Minor.Injuries  Total.Uninjured  
0                 0.000000         1.000000  
1                 0.000000         1.000000  
2                 0.000000         0.000000  
3                 1.000000         1.000000  
4                 0.000000         0.000000  
...                    ...              ...  
8233              0.000000         1.000000  
8234              0.000000         2.000000  
8235              0.111111         5.777778  
8236              0.000000         0.000000  
8237              1.000000         0.000000  

[8238 rows x 5 columns]

In [658]:
df = Aviation_data.groupby('Amateur.Built').agg({
    'Total.Fatal.Injuries': 'mean',
    'Total.Serious.Injuries': 'mean',
    'Total.Minor.Injuries': 'mean',
    'Total.Uninjured': 'mean'
}).reset_index()
df

Amateur.Built  Total.Fatal.Injuries  Total.Serious.Injuries  \
0       Missing              0.279308                0.036515   
1            No              0.580299                0.241409   
2           Yes              0.372861                0.227965   

   Total.Minor.Injuries  Total.Uninjured  
0              0.005766         1.723895  
1              0.312170         5.464327  
2              0.282950         0.910678